python/ray/data/impl/arrow_block.py

```diff
@@ -377,13 +379,24 @@ def gen():
                             next_row = None
                             break
 
-                accumulator = agg.init(next_key)
+                accumulators = [agg.init(next_key) for agg in aggs]
                 for r in gen():
-                    accumulator = agg.accumulate(accumulator, r)
+                    for i in range(len(aggs)):
+                        accumulators[i] = aggs[i].accumulate(
+                            accumulators[i], r)
                 if key is None:
-                    builder.add({agg.name: accumulator})
+                    builder.add({
+                        agg.name: accumulator
```

Comment:
Should we use the agg index instead of name in case there are conflicts?

Reply:
The aggregation names should be unique if the column names are unique (e.g. `mean(A)`), which should be guaranteed, and if there are any repeat aggregations on the same column(s), we'd just end up deduping duplicate accumulations. These column names are also user-facing, so we want them to have readable names if possible!

Outside of the scope of these PRs, I would like to changing the aggregation column naming scheme to something like `mean_A` to make it easier for users to programatically work with the aggregation results.

Reply:
Agree with renaming to agg_A etc. For the name, it can be user defined so I think it would be safer to not assume it's unique (or, we could raise a good error if it does collide?)

Reply:
>For the name, it can be user defined so I think it would be safer to not assume it's unique (or, we could raise a good error if it does collide?)

Ah, I just realized that you're referring to the aggregation name possibly not being unique, not the table column names, since aggregations can be user-defined. That's a great point!

I think that we can adopt the Pandas semantics of being forgiving here, and automatically dedupe by appending the duplicate count to the aggregation name.